pkg/controller/gcp/compute/network_test.go

```diff
@@ -0,0 +1,658 @@
+	corev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/meta"
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	"github.com/crossplaneio/crossplane-runtime/pkg/test"
+	"github.com/crossplaneio/crossplane/gcp/apis/compute/v1alpha1"
+	gcpv1alpha1 "github.com/crossplaneio/crossplane/gcp/apis/v1alpha1"
+)
+
+const testNetworkName = "test-network"
+const testNetworkDescription = "this is my test network!"
+const testGoogleProjectID = "test-project-id"
+const testProviderName = "test-provider"
+const testNamespace = "test-namespace"
+
+const testGCPCredentialsJSON = `
+{
+  "type": "service_account",
+  "project_id": "crossplane-test",
+  "private_key_id": "e7f8af9a43ff0513ea98616f410eb3d8dcd43159",
+  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKHITHERE!SkAgEAAoIBAQC2+Zn164pt1LHB\n0T2XN8LIqJWWtS2HT2EznQkq3UOzhhB3JlF3CA4NfpF7OgUhtIyQkGiTujM1U5rc\n8N7ZeDdWALugOX165ayqGZbQQZIgOVarzcXr8BsFTcfWQYkCSkqokdgkxHYvAKtD\nEyQorVOpMpc/hqKEuISxiDQUgFSDv4fgHHVcG/0oz7CB92Jaoiy2DLy4wYJQ0ikS\nc2CJBI4FkjDmsdQUvHBZXjxKxVvdN1NdeGsI34+YmzDgRgU1BunQJ/sfXq3iOHfg\n2DXwVkkSCriL8czH64fnAi3PAQg5uLSI4jCv/i5X/5MTslTL2QpAWqEYeeNFPRSI\nRIkqTuO1AgMBAAECggEARdCrLCg5Y3racGeUJpTCH6ojmaN7wbtKDVWH3/EZGfC3\na58VSmOqaP+4JtqWhqvLToWrQWK0k05ZMqA9h6DVDF2vOY12EWR9I1IsNz4tsVnv\ndhJHyKSOytFIxLmwu0z5e6RaYn0L0xdgQXGWRGcOGusQxs5qSFXXW53qsT/d/354\nXISUPJQX0r/GZ2Qd+o40HgP6L970sVJyC/+pTXPNgAHz/KamDTFkOUnFT03JPfEa\noYJs2j8+ur2ZONrOB9XL1nz2mX5U2y0wO6HjLNGwUH0fYAup8LjWMSzwkTUdZK3a\n+4lDU47IzITfrK0eWtErcqoOKqmpE8YtGx2drD63aQKBgQD5PFhO8VItjHDhAx0n\nexzB7BQ2wGCQz77RucYMzb9YFvL+DThSCxT+4VcwmqBBou4iTkICAyvCckHuIWmp\nDKP61jT8Lxe8ylblE5Mr2/YpAZCf408qwDzhkGqu9+xegc3TOy2H6dZtaD63578q\nL1JxBebV+jDhedabb6x6a47X/wKBgQC78OLgHUQqfbknwG3HpybBAd3nVLBsAPjF\nX/1Z0lrdEnq2SxrYLZK1PAPdq2dqaTDXRReXbrKVVbmDtVbvbbJHHmL7wqDtGw3o\ndhCn5PJDyqRmNNaJlCyfpIygNonwJ6Mji/KL78zsKqSed7hUurX+44po7tNhW4g0\n+5LJlOpkSwKBgHVSZLd0fZyXWaNyFc059MWMXc+QK2S/hjGvY30vdP0AgI36pyJn\nuRBp/uMQRpPfIH3hnMzZtUWpB0QTUmuTWbWAFB0KBgCGtvTKjCK/Plb7N5ssI14a\ndA4yLyYB6rBqJftjBgHOn3OM+MkVPDWdZQ9zGcBsL4pUEiPFI8nUy779AoGBAIM1\nh48jn4VyxKNmCr6KidnRrdRlIc4lV4iknUHqRBl3X0iJYrJuDrCMt744fjiiO8Du\nsau9UtxDPwMexRTfJsnUAxL1LFicvLcARlyQ0DpX7TKVzPm8AvNpc+weZ2rMoW6d\ns7XSsuRmWQtxVbWBrLHItcbQha7AsnUa9b3aQ1HFAoGBAMKmHpAsYqGr9nlztU51\niIwiE+jrNbuOhzgpP+fd5ZSaaTjHgh5JqdSGICjPGP7rYFFPItr6fCr6J2COpDXd\nyMxooGyoExA+NUJWwPOVTIXRPrW0GqSze7eAbdZgsp+DLj/IeJnhjS00CTYrEVMW\nSHmJ6XkR4JhwmS4dOz0aHJiH\n-----END PRIVATE KEY-----\n",
+  "client_email": "crossplane-test@crossplane-playground.iam.gserviceaccount.com",
+  "client_id": "110004009156833067752",
+  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+  "token_uri": "https://oauth2.googleapis.com/token",
+  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/monus-test%40crossplane-test.iam.gserviceaccount.com"
+}
```

Comment:
This looks suspiciously like real credentials. I've disabled the corresponding service account and deleted the corresponding private key.

If we do need a set of dummy credentials to test with we should ensure we don't leak too much information, and make it very obvious that it's dummy data.

Reply:
It's not, if you look long enough you'll see the `HITHERE!` in the private key. Also, such project doesn't exist and email in the `client_x509_cert_url` is different than `client_email`. I tried to keep the token generation function of Google working, so, that's why the structure had to seem similar. I'll try to manipulate it more while keeping the generation working.